cylc/flow/data_store_mgr.py

```diff
@@ -1584,15 +1868,18 @@ def prune_data_store(self):
         for tp_id in list(node_ids):
             if tp_id in self.n_window_nodes:
                 del self.n_window_nodes[tp_id]
-            if tp_id in self.n_window_edges:
-                del self.n_window_edges[tp_id]
             if tp_id in tp_data:
                 node = tp_data[tp_id]
             elif tp_id in tp_added:
                 node = tp_added[tp_id]
             else:
                 node_ids.remove(tp_id)
                 continue
+            self.n_window_edges.difference_update(set(node.edges))
```

Comment:
`set` methods can be called straight on iterables
```suggestion
            self.n_window_edges.difference_update(node.edges)
```